index.js

```diff
@@ -47,7 +48,7 @@ function lifecycle (pkg, stage, wd, opts) {
     validWd(wd || path.resolve(opts.dir, pkg.name), function (er, wd) {
       if (er) return reject(er)
 
-      if ((wd.indexOf(opts.dir) !== 0 || _incorrectWorkingDirectory(wd, pkg)) &&
+      if (((wd.indexOf(opts.dir) !== 0 && opts.dir !== path.join(wd, 'node_modules')) || _incorrectWorkingDirectory(wd, pkg)) &&
```

Comment:
Why is this change necessary? `opts.dir` is always the top level node modules directory of the package you're installing, while `wd` can be any directory under that. Does that not work for `pnpm`?

Reply:
but what about when executing scripts of the project? dir will be `project/node_modules`. wd will be `project`

Reply:
for dependencies it sure works. dir is `project/node_modules`, wd is `project/node_modules/dependency`

Reply:
npm calculates the prefix itself. `wd` works out to be the appropriate local prefix.

Reply:
I don't understand, npm doesn't use `npm/lifecycle` for the project's scripts? I had issues in pnpm when using `npm/lifecycle` because the value of dir was incorrect when running scripts in `project/package.json`

Reply:
Can you point me to a WIP branch on pnpm so I can help troubleshoot?

Reply:
There is no WIP branch currently. I did a fork of @npm/lifecycle with these and the [node-gyp](https://github.com/npm/lifecycle/pull/5) change. I revert this part for now and later when all the other things work, I will retry and create a new issue

Reply:
@zkochan `wd` is not the application's working directory: https://github.com/zkat/cipm/blob/latest/index.js#L149

We never give it something inside of the root of the project.

Furthermore, at the *root*, we use basically this code: https://github.com/zkat/cipm/blob/latest/lib/get-prefix.js to calculate what the "package root" is, even if you're, say, inside `lib`. Here's where we pass the prefix down: https://github.com/zkat/cipm/blob/latest/index.js#L36-L41

Using cipm 'cause it's a smaller codebase :)